tools/walletextension/api/staticOG/javascript.js

```diff
@@ -209,113 +236,123 @@ async function populateAccountsTable(document, tableBody, userID) {
-            statusArea.innerText = "Failed to add network"
-            return
-        }
-        statusArea.innerText = "Successfully joined Obscuro Gateway";
-        // show users an option to add another account and revoke userID
-        addAccountButton.style.display = "block"
-        addAllAccountsButton.style.display = "block"
         revokeUserIDButton.style.display = "block"
         accountsTable.style.display = "block"
         await populateAccountsTable(document, tableBody, userID)
-    })
+    }
 
-    addAccountButton.addEventListener(eventClick, async () => {
-        // check if we have userID and it is the correct length
-        if (!isValidUserIDFormat(userID)) {
-            statusArea.innerText = "\n Please join Obscuro network first"
-            joinButton.style.display = "block"
-            addAccountButton.style.display = "none"
+    async function displayCorrectScreenBasedOnMetamaskAndUserID() {
+        // check if we are on Obscuro Chain
+        if(await isObscuroChain()){
+            // check if we have valid userID in rpcURL
+            if (isValidUserIDFormat(userID)) {
+                return await displayConnectedAndJoinedSuccessfully()
+            }
         }
+        return displayOnlyJoin()
+    }
+
+    // load the current version
+    await fetchAndDisplayVersion();
 
-        await connectAccount()
+    await displayCorrectScreenBasedOnMetamaskAndUserID()
 
-        // Get an account and prompt user to sign joining with a selected account
-        const account = await provider.getSigner().getAddress();
-        if (account.length === 0) {
-            statusArea.innerText = "No MetaMask accounts found."
-            return
-        }
-        let authenticateAccountStatus = await authenticateAccountWithObscuroGateway(ethereum, account, userID)
-        //statusArea.innerText = "\n Authentication status: " + authenticateAccountStatus
-        accountsTable.style.display = "block"
-        await populateAccountsTable(document, tableBody, userID)
-    })
+    joinButton.addEventListener(eventClick, async () => {
+        // check if we are on an obscuro chain
+        if (await isObscuroChain()) {
+            userID = await getUserID()
+            if (!isValidUserIDFormat(userID)) {
+                statusArea.innerText = "Please remove existing Obscuro network from metamask and start again."
+            }
+        } else {
+            // we are not on an Obscuro network - try to switch
+            let switched = await switchToObscuroNetwork();
+            // error 4902 means that the chain does not exist
+            if (switched === 4902 || !isValidUserIDFormat(await getUserID())) {
+                // join the network
+                const joinResp = await fetch(
+                    pathJoin, {
+                        method: methodGet,
+                        headers: jsonHeaders,
+                    });
+                if (!joinResp.ok) {
+                    return
```

Comment:
Should something be logged to the console and/or user here ?

Reply:
Added console log and message to user to try again later.